datafusion/physical-expr/src/analysis.rs

```diff
@@ -184,24 +192,17 @@ fn shrink_boundaries(
         }
     });
     let graph_nodes = graph.gather_node_indices(&[expr.clone()]);
-    let (_, root_index) = graph_nodes.first().ok_or_else(|| {
-        DataFusionError::Internal("Error in constructing predicate graph".to_string())
-    })?;
-    let final_result = graph.get_interval(*root_index);
+    // Since propagation result was successful, the graph has at least one element.
+    // An empty check is also done at the outer scope, do not repeat it here.
+    let (_, root_index) = graph_nodes[0];
```

Comment:
I would prefer the internal error rather than this `[0]` which will panic. I know it is not supposed to happen, but avoiding a panic would be a nicer experience for users I think